src/Xamarin.Android.Build.Tasks/Utilities/XAJavaTypeScanner.cs

```diff
@@ -0,0 +1,167 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics;
+using System.IO;
+//using System.IO.MemoryMappedFiles;
+
+using Java.Interop.Tools.Cecil;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Mono.Cecil;
+using Xamarin.Android.Tools;
+
+namespace Xamarin.Android.Tasks;
+
+class JavaType
+{
+	public readonly TypeDefinition Type;
+	public readonly IDictionary<AndroidTargetArch, TypeDefinition>? PerAbiTypes;
```

Comment:
…and if you _are_ going to create a `ReadOnlyDictionary` wrapper, why isn't this typed as `ReadOnlyDictionary`?  Why keep this as `IDictionary`?

Reply:
Because one day I can change my mind and use a SortedReadOnlyDictionary? :)

Reply:
> Because one day I can change my mind and use a SortedReadOnlyDictionary? :-)

While a joke, that still doesn't make sense to me, because `ReadOnlyDictionary` doesn't "deep copy" the contents, it just wraps the modifying methods to throw `NotSupportedException`:

  * https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Collections/ObjectModel/ReadOnlyDictionary.cs

A `SortedReadOnlyDictionary` would make no sense.

Reply:
I can't protect the `TypeDefinition` (and don't want to), because it will be modified - I care about the **coupling** of architecture and the type definition. That's what is to be read only, and `ReadOnlyDictionary` provides that invariant. While `SortedReadOnlyDictionary` was a joke, the reason `IDictionary` is used is because I believe that's the best course of action in somewhat future-proof ("somewhat" because the field could in theory change type to `ICollection<SomethingDescribingAPerAbiTypeDefinition>`) API design. Users of `JavaType` don't need to know what concrete IDictionary implementation is used.